src/bin/cargo/commands/add.rs

```diff
@@ -242,7 +242,20 @@ fn parse_dependencies(config: &Config, matches: &ArgMatches) -> CargoResult<Vec<
         .flatten()
         .map(|c| (Some(c.clone()), None))
         .collect::<IndexMap<_, _>>();
+
     let mut infer_crate_name = false;
+
+    for (crate_name, _) in crates.iter() {
+        let crate_name = crate_name.as_ref().unwrap();
+
+        if let Some(toolchain) = crate_name.strip_prefix("+") {
+            anyhow::bail!(
+                "invalid character `+` in dependency name: \"+{toolchain}\"
```

Comment:
Found another. Sorry for being picky on these details. Ditto in `cargo install` as well.

```suggestion
                "invalid character `+` in dependency name: `+{toolchain}`
```

Reply:
Nah it's okay, I missed that too.